### PR TITLE
Fixes verifying the JWT on the installed and uninstalled events

### PR DESCRIPTION
**Pull request by bulinutza**
Atlassian has changed the way they sign their JWT for the `install` and `uninstall` hooks. From their [release notes](https://community.developer.atlassian.com/t/action-required-atlassian-connect-installation-lifecycle-security-improvements/49046):

> At installation time, the Connect app server and the host product exchange a shared secret used to create and validate JWT tokens for use in API calls. However, the very first install lifecycle hook is not signed because the secret has not been shared yet to the app server. This makes it difficult to verify that the lifecycle hook was genuinely requested from Atlassian, which may lead Connect apps to be vulnerable to Denial of Service (DoS) and/or Man in the Middle(MitM) attacks and opens the possibility for further exploits when an app secret can be specified by a MitM.
> 
> Before
> - Initial install hook is not signed
> - App upgrade install hooks are symmetrically signed with previously shared secret
> - Uninstall hook is also symmetrically signed with the shared secret
> - Initial and upgrade install hooks provide shared secret which is included in its request body
> - Shared secret will be used to create and validate JWT tokens for use in API calls between Atlassian and the app including any subsequent install hooks
> 
> After
> - All install(including upgrades)/uninstall hooks will be asymmetrically signed with RS256 (RSA Signature with SHA-256) algorithm
> - Public key will be provided via a CDN to verify the JWT token. `https://connect-install-keys.atlassian.com/${public_key_id}`
> - Initial and upgrade install hooks provide shared secret which is included in its request body
> - Shared secret will be used to create and validate JWT tokens for use in API calls between Atlassian and the app, excluding install and uninstall hooks

### Main changes in this PR
- always verify the JWT in the install and uninstall hooks, because as per the documentation they're always sent now, not only on update
- use the `force_asymmetric_verify` option to verify the JWT in these 2 hooks
- minor code improvements

### CODE REVIEW

File: lib/atlassian-jwt-authentication/version.rb [hunk 1]
```diff
@@ -1,7 +1,7 @@
 module AtlassianJwtAuthentication
   MAJOR_VERSION = "0"
   MINOR_VERSION = "9"
-  PATH_VERSION = "0"
+  PATH_VERSION = "1"
```
Comment (marinfr) [inline lib/atlassian-jwt-authentication/version.rb]: guess this was supposed to be patCh? 👀

File: lib/atlassian-jwt-authentication/filters.rb [hunk 2]
```diff
@@ -24,16 +24,16 @@ def on_add_on_installed
...
-          # Update request was issued to another plugin
-          render_forbidden
-          return false
-        end
-      else
+      if jwt_auth && jwt_auth.id != current_jwt_token.id
+        # Update request was issued to another plugin
+        render_forbidden
+        return false
+      elsif jwt_auth.nil?
```
Comment (marinfr) [inline lib/atlassian-jwt-authentication/filters.rb]: Inverting the if and elsif would make it easier to read imho:
```suggestion
      if jwt_auth.nil?
        self.current_jwt_token = JwtToken.new(jwt_token_params)
      elsif jwt_auth.id != current_jwt_token.id
        # Update request was issued to another plugin
        render_forbidden
        return false
```

**Review by marinfr — COMMENTED** [review]
Hope this fixes it! Just a few comments:

File: lib/atlassian-jwt-authentication/filters.rb (hunk 2 above)
Comment (bulinutza) [reply lib/atlassian-jwt-authentication/filters.rb]: You're right, I updated it.

File: lib/atlassian-jwt-authentication/version.rb (hunk 1 above)
Comment (bulinutza) [reply lib/atlassian-jwt-authentication/version.rb]: Indeed, fixed it now :-)

**Review by marinfr — APPROVED** [review]
Hope this fixes it!

**Review by skv-headless — APPROVED** [review]

**Review by pawelniewie — APPROVED** [review]

**Pull request merged**